core_lib/src/activeframepool.cpp

```diff
@@ -38,22 +35,27 @@ void ActiveFramePool::put(KeyFrame* key)
 
     Q_ASSERT(key->pos() > 0);
 
+    key->loadFile();
+
     auto it = mCacheFramesMap.find(key);
-    mCacheFramesList.push_front(key);
-    if (it != mCacheFramesMap.end())
+    const bool keyExistsInPool = (it != mCacheFramesMap.end());
+    if (keyExistsInPool)
     {
+        // move the keyframe to the front of the list, if the key already exists in frame pool
         mCacheFramesList.erase(it->second);
     }
+    mCacheFramesList.push_front(key);
     mCacheFramesMap[key] = mCacheFramesList.begin();
+
     key->addEventListener(this);
-    key->loadFile();
 
-    discardLeastUsedFrames();
-}
+    if (!keyExistsInPool)
+    {
+        mTotalUsedMemory += key->memoryUsage();
+        //qDebug() << "Total Memory:" << mTotalUsedMemory;
```

Comment:
Let's cleanup some of these comments.

Reply:
I thought it will be easy for someone who wants to debug this in the future. But anyway I can remove them.

Reply:
I mean it's simple debug statements, I don't think we should have them lying around, because it's convenient. At least I think they should be removed before merging.